netspeed/src/backend.c

```diff
@@ -207,6 +207,56 @@ get_ip_address_list (const char *iface_name,
     return list;
 }
 
+GSList*
+get_simple_ip_address_list (const char *iface_name,
+                            gboolean    incl_ipv6)
+{
+    GSList *list = NULL;
+    struct ifaddrs *ifaces;
+
+    if (getifaddrs (&ifaces) != -1) {
+        char ip[INET6_ADDRSTRLEN];
+
+        for (struct ifaddrs *iface = ifaces; iface != NULL; iface = iface->ifa_next) {
+            if (iface->ifa_addr == NULL)
+                continue;
+
+            if (!g_strcmp0 (iface->ifa_name, iface_name))
+            {
+                void *sinx_addr = NULL;
+
+                if (iface->ifa_addr->sa_family == AF_INET) {
+                    struct sockaddr_in ip4_addr;
+
+                    memcpy (&ip4_addr, iface->ifa_addr, sizeof (struct sockaddr_in));
+
+                    /* get network ip */
+                    sinx_addr = &ip4_addr.sin_addr;
+                    inet_ntop (iface->ifa_addr->sa_family, sinx_addr, ip, sizeof (ip));
+
+                    list = g_slist_prepend (list, g_strdup_printf ("%s", ip));
+                }
+
+                if (iface->ifa_addr->sa_family == AF_INET6 && incl_ipv6) {
+                    struct sockaddr_in6 ip6_addr;
+
+                    memcpy (&ip6_addr, iface->ifa_addr, sizeof (struct sockaddr_in6));
+
+                    /* get network ip */
+                    sinx_addr = &ip6_addr.sin6_addr;
+                    inet_ntop (iface->ifa_addr->sa_family, sinx_addr, ip, sizeof (ip));
+
+                    list = g_slist_prepend (list, g_strdup_printf ("%s", ip));
+                }
+            }
+        }
+        if (list != NULL)
+            list = g_slist_sort (list, (GCompareFunc) g_strcmp0);
```

Comment:
It will display mixed ipv4 and ipv6 addresses

Reply:
Hm, you might be right. In my cases it always showed IPv4 before IPv6, I guess because of the colon-notation instead of point-notation. What's your suggestion here? Remove it?

Reply:
> Hm, you might be right. In my cases it always showed IPv4 before IPv6, I guess because of the colon-notation instead of point-notation. What's your suggestion here? Remove it?

No, you can use another GCompareFunc.

Reply:
you can also use two lists...

Reply:
You are right. I checked it and the reason IPv4 mostly appear in front is because most IPv6 addresses start with >=2, while most IPv4 addresses start with <=2. Now I am thinking of 3 solutions:
1. Use different sort function (do you have a specific one in mind?)
2. Use 2 separate lists which are filled by 2 separate functions (`get_ipv4_address_list` and `get_ipv6_address_list`)
3. Modify `get_ip_address_list` to return structs that contain `ip`, `netmask`, and `scope` and then use this data in `netspeed.c`.

However, I would postpone option 3 because it is a much bigger code change and I don't know when I will find the time for this.

Reply:
To make your choice easier, consider that at most there is an ipv4 address and it's stored in binary format in device->ip variable (2nd solution)